bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,206 @@
+
+        match mode {
+            Mode::Client => {
+                read_ptr = server_to_client_ptr;
+                write_ptr = client_to_server_ptr;
+                config = &self.client_config;
+                conn = &mut self.client_conn;
+            }
+            Mode::Server => {
+                read_ptr = client_to_server_ptr;
+                write_ptr = server_to_client_ptr;
+                config = &self.server_config;
+                conn = &mut self.server_conn;
+            }
+        }
+
+        conn.set_blinding(Blinding::SelfService)?
+            .set_config(config.clone())?
+            .set_send_callback(Some(Self::send_cb))?
+            .set_receive_callback(Some(Self::recv_cb))?;
+        unsafe {
+            conn.set_send_context(write_ptr)?
+                .set_receive_context(read_ptr)?;
+        }
+
+        Ok(())
+    }
+
+    /// Handshake step for one connection
+    fn handshake_conn(&mut self, mode: Mode) {
+        let (conn, handshaked);
+        match mode {
+            Mode::Client => {
+                debug!("Client: ");
+                conn = &mut self.client_conn;
+                handshaked = &mut self.client_handshaked;
+            }
+            Mode::Server => {
+                debug!("Server: ");
+                conn = &mut self.server_conn;
+                handshaked = &mut self.server_handshaked;
+            }
+        }
+        if let Ready(res) = conn.poll_negotiate() {
+            res.unwrap();
+            *handshaked = true;
+        } else {
+            *handshaked = false;
+        }
+    }
+}
+
+impl TlsBenchHarness for S2nTls {
+    fn new() -> Self {
+        debug!("----- s2n-tls -----");
+        let mut new_struct = S2nTls {
+            client_to_server_buf: Box::pin(VecDeque::new()),
+            server_to_client_buf: Box::pin(VecDeque::new()),
+            client_config: Self::create_config(Mode::Client),
+            server_config: Self::create_config(Mode::Server),
+            client_conn: Connection::new_client(),
+            server_conn: Connection::new_server(),
+            client_handshaked: false,
+            server_handshaked: false,
+        };
+        new_struct.init_conn(Mode::Client).unwrap();
+        new_struct.init_conn(Mode::Server).unwrap();
+        new_struct
+    }
+
+    fn handshake(&mut self) {
+        if self.has_handshaked() {
+            return;
+        }
+        debug!("HANDSHAKE");
+        let mut round_trips = 2; // expect two round trips, second for server to see Finished message
+        while !self.has_handshaked() && round_trips > 0 {
+            self.handshake_conn(Mode::Client);
+            self.handshake_conn(Mode::Server);
+            round_trips -= 1;
+        }
+        debug!("HANDSHAKE DONE\n");
+    }
+
+    fn has_handshaked(&self) -> bool {
+        self.client_handshaked && self.server_handshaked
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::TlsBenchHarness;
```

Comment:
I think common rust style is to add a `use super:*` which imports everything in the parent module into your namespace, which lets you refer to functions in tests without having to use the `super` prefix.